File: airbyte-commons-auth/src/main/java/io/airbyte/commons/auth/support/UserAuthenticationResolver.java
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.commons.auth.support;

import io.airbyte.config.AuthenticatedUser;
import java.util.Optional;

/**
 * Interface for resolving user authentication attributes into an Airbyte User object.
 */
public interface UserAuthenticationResolver {

  AuthenticatedUser resolveUser(final String expectedAuthUserId);

  Optional<String> resolveRealm();

}

File: airbyte-commons-auth/src/main/kotlin/io/airbyte/commons/auth/support/UserAuthenticationResolver.kt
/*
 * Copyright (c) 2020-2026 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.commons.auth.support

import io.airbyte.config.AuthenticatedUser

/**
 * Interface for resolving user authentication attributes into an Airbyte User object.
 */
interface UserAuthenticationResolver {
  fun resolveUser(expectedAuthUserId: String): AuthenticatedUser

  fun resolveRealm(): String?
}
